backend/Squirrel.SqlService/Squirrel.SqlService.BLL/Services/ConsoleAppHub/ResultObserver.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Squirrel.SqlService.BLL.Models.ConsoleAppHub;
+
+namespace Squirrel.SqlService.BLL.Services.ConsoleAppHub;
+
+public class ResultObserver
+{
+    private readonly Dictionary<Guid, TaskCompletionSource<QueryResultTableDTO>> _pendingRequests = new();
```

Comment:
if it will be available to several (many) users we need some updates like

`private readonly ConcurrentDictionary<Guid, TaskCompletionSource<QueryResultTableDTO>> _pendingRequests = new();`